arch/arm/core/aarch32/cortex_m/mpu/arm_mpu.c

```diff
@@ -15,6 +15,16 @@
 #include <logging/log.h>
 LOG_MODULE_DECLARE(mpu);
 
+#if defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+#define MPU_NODEID DT_INST(0, arm_armv7m_mpu)
+#elif defined(CONFIG_ARMV8_M_BASELINE) || defined(CONFIG_ARMV8_M_MAINLINE)
+#define MPU_NODEID DT_INST(0, arm_armv8m_mpu)
+#endif
```

Comment:
I think here you should do similar to what you did for  nvic, and with a similar comment:

#if v8m_base || v8m_main
use v8m_mpu
#elif v6_m_v8_m_base || v7m_v8m_main
use v7m_mpu

Reply:
Ok, but I think it should be `arm_armv6m_mpu`,  I'd treat this the same way we do nvic.  I'm not aware of differences in the v6m and v7m nvic, but we give them different compatibles anyways.

Ok?

Reply:
sure :)

Reply:
Done.

Reply:
@galak I think you need to have:

```
#if defined(CONFIG_ARMV8_M_BASELINE) || defined(CONFIG_ARMV8_M_MAINLINE)
/* The order here is on purpose since ARMv8-M SoCs may define
 * CONFIG_ARMV6_M_ARMV8_M_BASELINE or CONFIG_ARMV7_M_ARMV8_M_MAINLINE
 *  so we want to check for ARMv8-M first.
 */
#define MPU_NODEID DT_INST(0, arm_armv8m_mpu)
#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
#define MPU_NODEID DT_INST(0, arm_armv7m_mpu)
#elif defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
#define MPU_NODEID DT_INST(0, arm_armv6m_mpu)
#endif

```